feat(config): add Swidget Z-Wave inserts

michaelkkehoe opened the pull request:
Introducing 4 new Swidget Z-Wave inserts

AlCalzone commented on packages/config/config/devices/0x0345/zw000uwajson:
```diff
@@ -0,0 +1,942 @@
+			]
+		},
+		{
+			"#": "13[0x64]",
```
same goes for the following partials

michaelkkehoe commented:
@AlCalzone Would you mind re-reviewing? Unfortunately I cannot find any error message pointing to what may be causing the linting tests to fail.

AlCalzone commented:
Will do after my vacation

AlCalzone commented on packages/config/config/devices/0x0345/templates/swidget_template.json:
```diff
@@ -0,0 +1,1504 @@
+{
+	"power_1_high_entry_value": {
+		// param 1
+		"label": "Power 1 High Entry Value",
+		"description": "Used to set the power value for Power 1 (the controlled load) that indicates entry into a Power 1 Low power state.",
```
Anyways, if we do this, all related parameters should be renamed in a similar fashion.

michaelkkehoe replied:
I'm removing the descriptions as they are causing problems. I am leaving the labels in a format the manufacturer would prefer (that matches their zwave documentation)

AlCalzone commented on packages/config/config/devices/0x0345/templates/swidget_template.json:
```diff
@@ -0,0 +1,1504 @@
+	"power_1_high_actions_control_host_on_entry": {
+		// param 14b2
+		"$import": "~/templates/master_template.json#base_enable_disable",
+		"label": "Power 1 High Actions: CONTROL HOST ON ENTRY",
+		"description": "If set, execute BASIC SET with “Power 1 High Entry SET Value” on host device on Power 1 High state entry."
```
```suggestion
		"label": "Load High Power Actions: Basic Set to Host (Entry)",
```

AlCalzone commented on packages/config/config/devices/0x0345/templates/swidget_template.json:
```diff
@@ -0,0 +1,1504 @@
+	},
+    "air_quality_level_1_index": {
+		// param 64
+		"label": "Air Quality Level 1 Index",
+		"description": "Used to set the index value representing the upper limit for the 'Clean' air state.",
```
```suggestion
		"label": "Air Quality Threshold: Clean",
```

michaelkkehoe replied:
These are the parameter names per the manufacturers zwave documentation. I would prefer to keep them in-sync.

AlCalzone commented on packages/config/config/devices/0x0345/templates/swidget_template.json:
```diff
@@ -0,0 +1,1504 @@
+	},
+    "air_quality_level_2_index": {
+		// param 65
+		"label": "Air Quality Level 2 Index",
```
What's the difference between this and the previous param?

michaelkkehoe replied:
These are the parameter names per the manufacturers zwave documentation. I would prefer to keep them in-sync.

AlCalzone commented on packages/config/config/devices/0x0345/templates/swidget_template.json:
```diff
@@ -0,0 +1,1504 @@
+	},
+    "air_quality_level_3_index": {
+		// param 66
+		"label": "Air Quality Level 3 Index",
+		"description": "Used to set the index value representing the upper limit for the 'Moderately Polluted' air state.",
```
```suggestion
		"label": "Air Quality: Moderately Polluted (Upper Threshold)",
```

AlCalzone replied:
Please apply this change in a similar fashion to the previous 2 params

michaelkkehoe replied:
These are the parameter names per the manufacturers zwave documentation. I would prefer to keep them in-sync.

AlCalzone commented on packages/config/config/devices/0x0345/templates/swidget_template.json:
```diff
@@ -0,0 +1,1504 @@
+	},
+	"air_quality_alert_level": {
+		// pram 67
+		"label": "Air Quality Alert Level",
+		"description": "Used to set the air quality level that will trigger Air Quality Alert State actions",
```
```suggestion
		"label": "Air Quality: Alert Threshold",
```

michaelkkehoe replied:
These are the parameter names per the manufacturers zwave documentation. I would prefer to keep them in-sync.

AlCalzone commented on packages/config/config/devices/0x0345/templates/swidget_template.json:
```diff
@@ -0,0 +1,1504 @@
+	},
+	"air_quality_alert_delay": {
+		// param 68
+		"label": "Air Quality Alert Delay",
+		"description": "Delay after detection of a level change before acting on change of air quality state",
```
Format this and the following params similar to the previous one. Descriptions all seem unnecessary.

michaelkkehoe replied:
These are the parameter names per the manufacturers zwave documentation. I would prefer to keep them in-sync.

AlCalzone commented on packages/config/config/devices/0x0345/templates/swidget_template.json:
```diff
@@ -0,0 +1,1504 @@
+	"air_quality_actions_control_associations_on_entry": {
+		// param 74b0
+		"$import": "~/templates/master_template.json#base_enable_disable",
+		"label": "Air Quality Actions: CONTROL ASSOCIATIONS ON ENTRY",
+		"description": "If set, send BASIC SET with 'Air Quality Alert Entry SET Value' to 'Air Quality' association group on entry into air quality alert state."
```
Please update similar to the High Power actions

AlCalzone commented on packages/config/config/devices/0x0345/templates/swidget_template.json:
```diff
@@ -0,0 +1,1504 @@
+		// param 80
+		"label": "Temperature High Entry Level",
+		"description": "Used to set the temperature value that indicates entry into a high/over temperature state.",
+		"unit": "Tenths of a Degree",
```
```suggestion
		"unit": "0.1 °C",
```
or is it Fahrenheit?

michaelkkehoe replied:
It can be either. Depending on if it's set to measure in C or F

AlCalzone commented on packages/config/config/devices/0x0345/templates/swidget_template.json:
```diff
@@ -0,0 +1,1504 @@
+	},
+	"led_invert": {
+		// param 154 b0
+		"label": "LED invert",
```
```suggestion
		"label": "LED Indicator",
```

michaelkkehoe replied:
Fixed this to be clearer

AlCalzone commented on packages/config/config/devices/0x0345/templates/swidget_template.json:
```diff
@@ -0,0 +1,1504 @@
+	},
+	"switch_host_control_flags_3_way_mode": {
+		// param 168b0
+		"label": "Switch Host Control Flags: 3-Way Mode",
```
I think just 
```suggestion
		"label": "3-Way Mode",
```
is enough

michaelkkehoe replied:
Fixed

AlCalzone commented on packages/config/config/devices/0x0345/templates/swidget_template.json:
```diff
@@ -0,0 +1,1504 @@
+	},
+	"switch_host_control_flags_on_off_invert": {
+		// param 168b1
+		"label": "Switch Host Control Flags: On/Off Invert",
```
```suggestion
		"label": "Invert Host State",
```
or
```suggestion
		"label": "Invert Host State And Commands",
```

AlCalzone commented on packages/config/config/devices/0x0345/templates/swidget_template.json:
```diff
@@ -0,0 +1,1504 @@
+		"maxValue": 1,
+		"defaultValue": 0,
+		"options": [
+			{
+				"label": "host output is reported on when energized, off when de-energized. Commands to host are not inverted.",
+				"value": 0
+			},
+			{
+				"label": "host output is reported off when energized, on when de-energized. Commands to host are inverted to match.",
+				"value": 1
+			}
```
```suggestion
			{
				"label": "Not inverted",
				"value": 0
			},
			{
				"label": "Inverted",
				"value": 1
			}
```

AlCalzone reviewed (changes requested):
The lint error is just about formatting. Unfortunately, our auto-fix action is broken, so you'll have to do it manually using
```
yarn run lint:prettier:fix
```

Aside from that, please review all descriptions and see if they are really necessary. Try to fit the information into the label (short) where possible. I've highlighted a couple of things that can/should be applied throughout the file.

michaelkkehoe commented:
@AlCalzone I've removed the descriptions as the manufacturer would prefer users to read their zwave manual. I've addressed most of your call-outs. I am trying to stay consistent with their zwave documentation as much as possible.

AlCalzone commented:
We often rename labels for clarity and brevity, even if they don't match the manual then (lots of those are terrible). In this case I can live with the labels though. Fixed a typo I saw and changed the units to be consistent with how we usually write them.
